### ISSUE TITLE
False Positive useless assignment to local variable 

### ISSUE THREAD

**Issue by Dixtosa** [SonarSource/sonar-dotnet#3115]
### Description

S1854 fails on tuples.

### Repro steps
This generates warning:
```
public object WellNamedMethod()
{
            bool wellNamedBoolean = 2 == 2 * 1;
            return (new int[0] { }, wellNamedBoolean);
}
```

This does not:
```
public object WellNamedMethod()
{
            bool wellNamedBoolean = 2 == 2 * 1;
            return wellNamedBoolean;
}
```

### Expected behavior

I expect the warning to not show up. 

### Actual behavior

The warning is "remove this useless assignment to local variable".

### Known workarounds

-

### Related information

* SonarLint for Visual Studio 2017 4.17.0.14702
* Visual Studio Pro 2017

**Comment by andrei-epure-sonarsource** [comment]
thanks @Dixtosa , this looks like a duplicate of https://github.com/SonarSource/sonar-dotnet/issues/3094

this fix for #3094 is on master and will be in the next release
